libadikted/Makefile.win

```diff
@@ -1,106 +1,99 @@
-
-bulcommn.o: bulcommn.c
-	$(CC) -c bulcommn.c -o bulcommn.o $(CFLAGS)
-
-dernc.o: dernc.c
-	$(CC) -c dernc.c -o dernc.o $(CFLAGS)
-
-draw_map.o: draw_map.c
-	$(CC) -c draw_map.c -o draw_map.o $(CFLAGS)
-
-graffiti.o: graffiti.c
-	$(CC) -c graffiti.c -o graffiti.o $(CFLAGS)
-
-graffiti_font.o: graffiti_font.c
-	$(CC) -c graffiti_font.c -o graffiti_font.o $(CFLAGS)
-
-lev_column.o: lev_column.c
-	$(CC) -c lev_column.c -o lev_column.o $(CFLAGS)
-
-lev_data.o: lev_data.c
-	$(CC) -c lev_data.c -o lev_data.o $(CFLAGS)
-
-lev_files.o: lev_files.c
-	$(CC) -c lev_files.c -o lev_files.o $(CFLAGS)
-
-lev_script.o: lev_script.c
-	$(CC) -c lev_script.c -o lev_script.o $(CFLAGS)
-
-lev_things.o: lev_things.c
-	$(CC) -c lev_things.c -o lev_things.o $(CFLAGS)
-
-memfile.o: memfile.c
-	$(CC) -c memfile.c -o memfile.o $(CFLAGS)
-
-obj_actnpts.o: obj_actnpts.c
-	$(CC) -c obj_actnpts.c -o obj_actnpts.o $(CFLAGS)
-
-obj_column.o: obj_column.c
-	$(CC) -c obj_column.c -o obj_column.o $(CFLAGS)
-
-obj_column_def.o: obj_column_def.c
-	$(CC) -c obj_column_def.c -o obj_column_def.o $(CFLAGS)
-
-obj_column_per.o: obj_column_per.c
-	$(CC) -c obj_column_per.c -o obj_column_per.o $(CFLAGS)
-
-obj_slabs.o: obj_slabs.c
-	$(CC) -c obj_slabs.c -o obj_slabs.o $(CFLAGS)
-
-obj_things.o: obj_things.c
-	$(CC) -c obj_things.c -o obj_things.o $(CFLAGS)
-
-xcubtxtr.o: xcubtxtr.c
-	$(CC) -c xcubtxtr.c -o xcubtxtr.o $(CFLAGS)
-
-xtabdat8.o: xtabdat8.c
-	$(CC) -c xtabdat8.c -o xtabdat8.o $(CFLAGS)
-
-xtabjty.o: xtabjty.c
-	$(CC) -c xtabjty.c -o xtabjty.o $(CFLAGS)
-
-msg_log.o: msg_log.c
-	$(CC) -c msg_log.c -o msg_log.o $(CFLAGS)
-
-arr_utils.o: arr_utils.c
-	$(CC) -c arr_utils.c -o arr_utils.o $(CFLAGS)
-
-lbfileio.o: lbfileio.c
-	$(CC) -c lbfileio.c -o lbfileio.o $(CFLAGS)
-
-adikted_private.res: adikted_private.rc 
-	$(WINDRES) -i adikted_private.rc --input-format=rc -o adikted_private.res -O coff 
+# Makefile created by KeeperFX/dkfans team
+
+# To create a debug executable, run with the command:
+# make DEBUG (THIS IS NOT IMPLEMENTED)
+
+# Instructions:
+# To create adikted.dll, run:
+# make all
+# To create adikited.so, run:
+# make all BUILD_FOR_LINUX=1
+# To clean up the built files (objects, library, and resource file) run:
+# make clean
+
+# Tested and working with:
+# - MSYS2 (MinGW 32-bit) - make
+# - MSYS2 (MinGW 32-bit) - mingw32-make
+# - MinGW (MSYS) - make
+
+# Executable files extension on host environment
+ifneq (,$(findstring Windows,$(OS)))
+  CROSS_EXEEXT = .exe
+else
+  CROSS_EXEEXT =
+  CROSS_COMPILE = i686-w64-mingw32-
+endif
+# Executable files extension on target environment (
+#BUILD_FOR_LINUX = 1
+ifneq (,$(BUILD_FOR_LINUX))
+  LIBEXT = .so
+  OTHERLIBEXT = .dll
+else
+  LIBEXT = .dll
+  OTHERLIBEXT = .so
+endif
+# Names of utility commands
+CPP                = $(CROSS_COMPILE)g++$(CROSS_EXEEXT) -std=c89
+CC                 = $(CROSS_COMPILE)gcc$(CROSS_EXEEXT) -std=c89 -Wall -Wextra -Wno-pedantic -Wno-conversion -Wno-traditional-conversion -Wno-sign-compare
```

Comment:
`Makefile.win` has these extra flags, and those on the lines below.

Which ones are wanted/needed?

Reply:
So when I initially made libadikted compile with GCC 9.2.0 my goal, and well still is to eventually make it compile to WASM and given the age of the code base I figured that starting to enable warnings was a good idea, so these extra flags are a recent change by me. That said the changes I made to fix warnings are untested. If its decided to remove the extra flags perhaps consider reverting my changes that went along with that, see commits:

- 5cb98708542204800cce24ba8af9282ac24ad122
- a667154e0500aaf8314da092c54d783edee50658
- e624588d2a0ff1730aca42e4d103022120b8298f
- e4561cd6ae0a3184346f30d9e6b3c3e8cee65a3a
- 233648b67a30428dbc7b69aea5c42b5c29d3ea0e
- a784b451948f2d44d35522fc76bcd6ce402a71ea
- d3b1f8a26f144103867df92fef06132ccaa15a51
- 52b68811c12ae815bb92d50f1eb361f50cb22ecc

Reply:
OK, that sounds good to me (keeping the warnings you added). The only difference I can see is that, surprise, warnings are generated. They will go in the merged file.